Data/ByteString/Short/Internal.hs

```diff
@@ -198,12 +199,30 @@ length (SBS _ len) = len
 null :: ShortByteString -> Bool
 null sbs = length sbs == 0
 
--- | /O(1)/ 'ShortByteString' index (subscript) operator, starting from 0. 
+-- | /O(1)/ 'ShortByteString' index (subscript) operator, starting from 0.
 index :: ShortByteString -> Int -> Word8
 index sbs i
   | i >= 0 && i < length sbs = unsafeIndex sbs i
   | otherwise                = indexError sbs i
 
+-- | /O(1)/ 'ShortByteString' index, starting from 0, that returns 'Just' if:
+--
+-- > 0 <= n < length bs
+--
+-- @since 0.10.10.1
+indexMaybe :: ShortByteString -> Int -> Maybe Word8
+indexMaybe sbs i
+  | i >= 0 && i < length sbs = Just $ unsafeIndex sbs i
```

Comment:
Very minor point: do we want to return `unsafeIndex sbs i` lazily?

Reply:
Good catch! I wonder whether it would be better to be more strict in the other versions too!

Reply:
FWIW, I agree this should be strict.